contracts/DebtSettler.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+
+import {ILPool} from "../interfaces/IPool.sol";
+
+import "../libraries/Errors.sol";
+
+contract DebtSettler is ReentrancyGuard {
+    using SafeERC20 for IERC20Metadata;
+    using EnumerableMap for EnumerableMap.AddressToUintMap;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    ILPool public immutable pool;
+    IERC20Metadata public immutable asset;
+
+    // Keep track of the debt for each borrower
+    EnumerableMap.AddressToUintMap internal _records;
+    // Keep track of borrowers
+    EnumerableSet.AddressSet internal _borrowers;
+
+    error InvalidPool();
+    error UnknownSender();
+    error ZeroDebt();
+
+    constructor(ILPool _pool) {
+        if (address(_pool) == address(0)) revert Errors.ZeroAddress();
+        if (_pool.expired()) revert InvalidPool();
+
+        pool = _pool;
+        asset = IERC20Metadata(_pool.asset());
+    }
+
+    modifier onlyPool() {
+        if (msg.sender != address(pool)) revert UnknownSender();
+        _;
+    }
+
+    function build(uint _amount) external onlyPool nonReentrant {
+        uint _totalDebt = 0;
+        uint _length = _borrowers.length();
+
+        // Prevent double call to _debt()
+        uint[] memory _debts = new uint[](_length);
+
+        // Get the total debt for all borrowers
+        for (uint i = 0; i < _length; i++) {
+            _debts[i] = pool.debt(_borrowers.at(i));
+
+            _totalDebt += _debts[i];
+        }
+
+        if (_totalDebt <= 0) revert ZeroDebt();
```

Comment:
I'ld return here, instead of revert just to prevent a possible failure in harvest =)